From 78f7a1800f18b567204a58b68acc805eed0c86e6 Mon Sep 17 00:00:00 2001
From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Date: Mon, 9 Jan 2023 05:18:16 +0000
Subject: [PATCH 04/11] fuse: Convert fuse_flush() to use
 file_check_and_advance_wb_err()

As with fsync, use the newer file_check_and_advance_wb_err() instead
of filemap_check_errors().

Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
---
 fs/fuse/file.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/fs/fuse/file.c b/fs/fuse/file.c
index 875314ee6..7174646dd 100644
--- a/fs/fuse/file.c
+++ b/fs/fuse/file.c
@@ -500,11 +500,10 @@ static int fuse_flush(struct file *file, fl_owner_t id)
 	fuse_sync_writes(inode);
 	inode_unlock(inode);
 
-	err = filemap_check_errors(file->f_mapping);
+	err = file_check_and_advance_wb_err(file);
 	if (err)
 		return err;
 
-	err = 0;
 	if (fm->fc->no_flush)
 		goto inval_attr_out;
 
-- 
2.39.0.rc2.1.gbd5df96b79

